src/node/rpc/memberfrontend.h

```diff
@@ -66,6 +66,19 @@ namespace ccf
            nodes->put(id, *info);
            return true;
          }},
+        // retire a node
+        {"retire_node",
+         [this](Store::Tx& tx, const nlohmann::json& args) {
+           const auto id = args;
+           auto nodes = tx.get_view(this->network.nodes);
+           auto info = nodes->get(id);
+           if (!info)
+             throw std::logic_error("Node does not exist.");
+           NodeInfo retired_info;
+           retired_info.status = NodeStatus::RETIRED;
```

Comment:
This will delete all the existing information about the retired node, which I guess it's probably fine. Would it be better to do `info.status = NodeStatus::RETIRED` instead?

Reply:
I thought this was the expected behavior, according to the scripts used on bob. I can keep all the data and ignore retired nodes when checking for duplicates (during add_node).